src/Compilers/CSharp/Portable/Symbols/TypeSymbolWithAnnotations.cs

```diff
@@ -20,6 +20,41 @@ internal enum NullableAnnotation : byte
         Nullable,     // Explicitly set by flow analysis
     }
 
+    /// <summary>
+    /// The nullable state of an rvalue computed in <see cref="NullableWalker"/>.
+    /// When in doubt we conservatively use <see cref="NullableFlowState.NotNull"/>
+    /// to minimize diagnostics.
+    /// </summary>
+    internal enum NullableFlowState : byte
+    {
+        NotNull,
+        MaybeNull
+    }
+
+    /// <summary>
+    /// A type and its corresponding flow state resulting from evaluating an rvalue expression.
+    /// </summary>
+    [DebuggerDisplay("{GetDebuggerDisplay(), nq}")]
+    internal readonly struct TypeWithState
+    {
+        public TypeSymbol Type { get; }
```

Comment:
>Type [](start = 26, length = 4)

Nit: we should use `TypeSymbol` for this to be consistent with `TWSA`

Reply:
I don't want to be consistent with TSWA.  See [https://github.com/dotnet/roslyn/issues/33736](https://github.com/dotnet/roslyn/issues/33736) which I expect to do in a separate PR.

---
In reply to: [260939120](https://github.com/dotnet/roslyn/pull/33648#discussion_r260939120) [](ancestors = 260939120)